src/Ocelot.Provider.Kubernetes/Kube.cs

```diff
@@ -26,28 +30,39 @@ public class Kube : IServiceDiscoveryProvider
         _logger = factory.CreateLogger<Kube>();
         _kubeApi = kubeApi;
         _serviceBuilder = serviceBuilder;
-        _services = new();
     }
 
     public virtual async Task<List<Service>> GetAsync()
     {
-        var endpoint = await _kubeApi
-            .ResourceClient(client => new EndPointClientV1(client))
-            .GetAsync(_configuration.KeyOfServiceInK8s, _configuration.KubeNamespace);
+        var endpoint = await Retry.OperationAsync(GetEndpoint, Ensure, logger: _logger);
 
-        _services.Clear();
-        if (endpoint?.Subsets.Count != 0)
+        if (Ensure(endpoint))
```

Comment:
You could invert the if, but that's a question of taste

Reply:
Makes sense, but you mean readability issue?... I pass `Ensure` delegate to `Retry` method. so I can't negate delegate ))
But I can rename the method to `NotSure` or `Check` or `IsInvalid`.
Or I can negate `bool` constants inside the method, but it won't fix readability issue. So, I'd prefer to rename to `Check` without negation. Please note that according to `OperationAsync` definition "Retry an operation when **predicate is true**". So, I don't want to re-define `OperationAsync`.